File: 1507-reformat-date/1507-reformat-date.py
class Solution:
    def reformatDate(self, date: str) -> str:
        dates = date.split(' ')
        dates[0] = dates[0].replace("st","").replace("rd","").replace("th","").replace("nd", "")
        if len(dates[0]) == 1:
            dates[0] = "0"+dates[0]
        monthDict = {
            "Jan": "01",
            "Feb": "02",
            "Mar": "03",
            "Apr": "04",
            "May": "05",
            "Jun": "06",
            "Jul": "07",
            "Aug": "08",
            "Sep": "09",
            "Oct": "10",
            "Nov": "11",
            "Dec": "12"
        }
        dates[1] = monthDict[dates[1]]
        return "-".join(dates[::-1])
        

File: 1507-reformat-date/1507-reformat-date.java
class Solution {
    public String reformatDate(String date) {
        StringBuilder sb = new StringBuilder();
        sb.append(date.substring(date.length()-4)).append("-");
        String m = date.length()==13 ? date.substring(5,8) : date.substring(4,7);
        switch(m){
             case "Jan": sb.append("01-"); break;
             case "Feb": sb.append("02-"); break;
             case "Mar": sb.append("03-"); break;
             case "Apr": sb.append("04-"); break;
             case "May": sb.append("05-"); break;
             case "Jun": sb.append("06-"); break;
             case "Jul": sb.append("07-"); break;
             case "Aug": sb.append("08-"); break;
             case "Sep": sb.append("09-"); break;
             case "Oct": sb.append("10-"); break;
             case "Nov": sb.append("11-"); break;
             case "Dec": sb.append("12-"); break;
         }
        String day = date.length()==13 ? date.substring(0,2) : "0"+date.substring(0,1);
        sb.append(day);
        return sb.toString();
    }
}